rskj-core/src/main/java/co/rsk/peg/BridgeStorageProvider.java

```diff
@@ -460,21 +461,28 @@ public void setPendingFederation(PendingFederation federation) {
     /**
      * Save the pending federation
      */
-    public void savePendingFederation() {
+    public void savePendingFederation() throws IOException {
         if (shouldSavePendingFederation) {
-            RepositorySerializer<PendingFederation> serializer = BridgeSerializationUtils::serializePendingFederationOnlyBtcKeys;
-
             if (activations.isActive(RSKIP123)) {
                 // we only need to save the standard part of the fed since the emergency part is constant
                 saveStorageVersion(
                     PENDING_FEDERATION_FORMAT_VERSION.getKey(),
                     STANDARD_MULTISIG_FEDERATION.getFormatVersion()
                 );
-                serializer = BridgeSerializationUtils::serializePendingFederation;
             }
+            savePendingFederation(pendingFederation);
+        }
+    }
 
-            safeSaveToRepository(PENDING_FEDERATION_KEY, pendingFederation, serializer);
+    /**
+     * Save the pending federation
+     */
+    private void savePendingFederation(PendingFederation pendingFederation) throws IOException {
```

Comment:
Good job on splitting this into two separate methods, but the fact that both have the same name can make it a bit confusing I think. Another developer wouldn't know which one to use probably.